### PR TITLE
AP_Arming: skip AHRS vs GPS location check if GPS is disabled

### PR DESCRIPTION
**Pull request by rmackay9**
This PR clears up some false-positive arming failures when using Non-GPS navigation (e.g. the [ModalAI VOXL camera](https://ardupilot.org/copter/docs/common-modalai-voxl.html)).  The troublesome messages are:

- GPS and AHRS differ by 1.5e+07m
- High GPS HDOP (Copter only)

The changes are:

1. Skip the AP_Arming comparison of AHRS vs the primary GPS if no GPSs are enabled (e.g. AP_GPS::num_sensors() == 0)
2. Skip the Copter specific GPS HDOP arming check if no GPSs are enabled (e.g. AP_GPS::num_sensors() == 0)

BTW, I was concerned about the difference between "AP_GPS::num_sensors()" and the number configured (e.g. GPS_TYPE, GPS_TYPE2) but the existing AP_GPS::first_unconfigured_gps() check catches cases where the user has set GPS_TYPE, GPS_TYPE2 but the drivers haven't found the GPS.  It also should not be an issue if only the 2nd GPS is configured (e.g. GPS_TYPE = 0, GPS_TYPE2 = x).

This PR also includes two small formatting changes:

3. AP_Arming indentation fix (surely left over from a previous PR that wanted to keep the diff to a minimum)
4. Replaced "GPS: waiting for home" with "AHRS: waiting for home" which is more accurate.

This PR resolves issues https://github.com/ArduPilot/ardupilot/issues/22383, https://github.com/ArduPilot/ardupilot/issues/14661 and https://github.com/ArduPilot/ardupilot/issues/14841

Below are some screen shots of testing on a real vehicle.
pr-ahrs-vs-gps-before-vs-after

### CODE REVIEW

**Comment by MallikarjunSE** [comment]
Hey @rmackay9 
There is lot of EKF3 variance when GPS is not connected to AP. I have few Cubes I use for bench testing, I get yaw inconsistency all the time even with cube orange internal compass. 
This behavior is recent and I never had this. When GPS is plugged in(without fix) there is no inconsistency.

File: libraries/AP_Arming/AP_Arming.cpp [hunk 1]
```diff
@@ -602,13 +602,15 @@ bool AP_Arming::gps_checks(bool report)
...
 
         // check AHRS and GPS are within 10m of each other
-        const Location gps_loc = gps.location();
-        Location ahrs_loc;
-        if (AP::ahrs().get_location(ahrs_loc)) {
-            const float distance = gps_loc.get_distance(ahrs_loc);
-            if (distance > AP_ARMING_AHRS_GPS_ERROR_MAX) {
-                check_failed(ARMING_CHECK_GPS, report, "GPS and AHRS differ by %4.1fm", (double)distance);
-                return false;
+        if (gps.num_sensors() > 0) {
```
Comment (peterbarker) [inline libraries/AP_Arming/AP_Arming.cpp:605]: This obviously loosens the checks, which is good.

It doesn't address one of the issues directly - where we have GPS sensors, but the EKF is currently configured not to use them (e.g. via ek3-source-sets).  In this situation the user may or may not want to arm (depending on their use case).  If that use case is transitioning from a GPS-denied env where it is being armed to a GPS environment then they want most of the checks in this function but not this one.

**Review by peterbarker — APPROVED** [review]
LGTM

File: libraries/AP_Arming/AP_Arming.cpp (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_Arming/AP_Arming.cpp:605]: @peterbarker,

Txs very much for the review and approval.

I don't quite understand the last sentence but I think you're saying we still have issues when the GPS is configured but the vehicle is being armed in a non-GPS environment.  I totally agree and I hope we can fix those in a follow-up PR at some point.

**Comment by rmackay9** [comment]
Hi @MallikarjunSE,

Txs for the report.  This is probably not the best place to discuss this issue though.  Could you perhaps create a new topic in the support area and if possible include logs from 4.4-dev and earlier working versions?

Although I haven't seen it myself, it's certainly possible we have an issue so I'm happy to help investigate.

**Comment by rmackay9** [comment]
Thanks for the review and approval, merged!

**Comment by rmackay9** [comment]
This is included in 4.3.3-rc1

**Pull request merged**